### PR TITLE
Setting up testing infrastructure

### PR DESCRIPTION
**Pull request by Marketionist**
**This pull request adds front-end tests infrastructure**

Details:
- [x] adds ```.editorconfig```, ```.jscsrc```, ```.jshintrc```
- [x] updates ```.gitignore```
- [x] adds ```browserslist```
- [x] adds ```gulpfile.js```
- [x] adds ```package.json```
- [x] adds ```test_requirements/runserver.sh```
- [x] updates ```tox.ini```
- [x] updates ```.travis.yml```
- [x] adds Sauce Labs badge
- [x] adds basic configuration for unit tests
- [x] adds basic configuration for integration tests

### CODE REVIEW

**Comment by Marketionist** [comment]
This PR will be merged later - waiting for review + till Travis builds to be finished /cc @FinalAngel @mkoistinen

File: .travis.yml [hunk 1]
```diff
@@ -3,36 +3,59 @@ language: python
...
+  #  - TOXENV=py27-dj17-cms30-postgres
+  #  - TOXENV=py27-dj17-cms31-mysql
+  #  - TOXENV=py27-dj17-cms31-postgres
+
+cache:
+  directories:
+    - $HOME/.wheelhouse
+
+before_install:
+  - npm install -g
```
Comment (vxsx) [inline .travis.yml]: install what?

Comment (Marketionist) [inline .travis.yml]: Should be ```- npm install -g npm``` - fixed

**Comment by FinalAngel** [comment]
LGTM

**Pull request merged**